### PR TITLE
Implemented event_decoders_test.go

### PR DESCRIPTION
**Pull request by x-senpai-x**
On line 41 I used a different value than what was used in python test suite due to discrepancy between the expected values in the Go and Python implementations for U256 .

### CODE REVIEW

File: athena_abi/event_decoders_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                default:
+                    t.Errorf("Unexpected type for key %s: %T", key, expectedValue)
+                }
+            }
+        })
+    }
+}
+
+// Helper function to parse big integers
+func mustParseBigInt(s string, base int) *big.Int {
```
Comment (Sh0g0-1758) [inline athena_abi/event_decoders_test.go:85]: Move these helper functions to uitls.go

File: athena_abi/event_decoders_test.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    tests := []struct {
+        name       string
+        abiEvent   *AbiEvent
+        eventData  []*big.Int
+        eventKeys  []*big.Int
+        expected   map[string]interface{}
+    }{
+        {
+            name: "Approval Event",
+            abiEvent: NewAbiEvent(
```
Comment (Sh0g0-1758) [inline athena_abi/event_decoders_test.go:20]: How did you get this ? In the original code, I saw that they have loaded an abi for the testing purpose.

Comment (Sh0g0-1758) [reply athena_abi/event_decoders_test.go:20]: Okay, Lets wait for core.go to be implemented. Then you can use from_json function from it.

**Review by Sh0g0-1758 — CHANGES_REQUESTED** [review]
In general, should be good to merge with a couple of nits.

File: athena_abi/event_decoders_test.go (hunk 2 above)
Comment (x-senpai-x) [reply athena_abi/event_decoders_test.go:20]: @Sh0g0-1758 Do I need to incorporate starknetabifromjson?

**Pull request closed without merging**